site-src/geps/gep-713.md

```diff
@@ -0,0 +1,409 @@
+resource in any namespace but it can only apply to requests from the same
+namespace if the target is in a different namespace.
+
+At the other extreme, policy can be used to apply to requests from a specific
+workload to a backend in another namespace. A route can be used to intercept
+these requests and split them between different backends (foo-a and foo-b in
+this case).
+
+![Complex Mesh Example](images/713-mesh-complex.png)
+
+### Policy TargetRef API
+
+Each Policy resource should include a single `targetRef` field. It must not
+target more than one resource at a time, but it can be used target larger
+resources such as Gateways or Namespaces that may apply to multiple child
+resources. This field should have the following structure:
+
+```go
+// PolicyTargetReference identifies an API object to apply policy to.
+type PolicyTargetReference struct {
+    // Group is the group of the target resource.
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=253
+    Group string `json:"group"`
+
+    // Kind is kind of the target resource.
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=253
+    Kind string `json:"kind"`
+
+    // Name is the name of the target resource.
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=253
+    Name string `json:"name"`
+
+    // SectionName is the name of a section within the target resource. When 
+    // unspecified, this targets the entire resource. In the following
+    // resources, SectionName is interpreted as the following:
+    // * Gateway: Listener Name
+    // * Route: Rule Name
+    // * Service: Port Name
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=253
+    // +optional
+    SectionName string `json:"sectionName,omitempty"`
+
+    // Namespace is the namespace of the referent. When unspecified, the local
+    // namespace is inferred. Even when policy targets a resource in a different
+    // namespace, it may only apply to traffic originating from the same 
+    // namespace as the policy.
+    //
+    // Support: Extended
+    //
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=253
+    // +optional
+    Namespace string `json:"namespace,omitempty"`
+    
+    // ClassName is the name of the class this policy should apply to. When
+    // unspecified, the policy will apply to all classes that support it.
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=253
+    // +optional
+    ClassName string `json:"className,omitempty"`
+}
+```
+
+### Status
+A new `Policies` field will be added to status on Gateways and Routes. This will
```

Comment:
To clarify, this is only intended to include policy that has been directly attached to the resource, not any it might inherit through other means. I think that's where the real scale issues come into play and the proposed kubectl plugin would be helpful. Of course only showing directly attached policies may lead to more confusion than it's worth.

Reply:
Good point, you probably wouldn't actually have thousands then, unless you have extensive usage of sectionName.

Given the missing status on the 4/6 of the attachment points and complexity around multiple controllers, we may consider just having the plugin, at least initially? I cannot really think how it can work with multiple controllers

Reply:
Yeah, I think you're right. I've summarized this and recommended that we may want to consider adding this at a later point but should focus on a kubectl plugin first.